File: samples/cucumber-demo/features/step_definitions/search.steps.ts
import { Given, When, Then } from '@cucumber/cucumber';

Given('I have {int} cucumbers', function (count: number) {
  this.count = count;
});

When('I search for {string}', function (query: string) {
  this.query = query;
});

Then(/I should see "([^"]+)" in results/, function (expected: string) {
  if (!expected) throw new Error('expected missing');
});

File: samples/behave-demo/features/steps/search_steps.py
from behave import given, when, then


@given('I have {n} cucumbers')
def step_given_cukes(context, n):
    context.count = int(n)


@when('I search for {query}')
def step_when_search(context, query):
    context.query = query


@then('I should see "{message}" in results')
def step_then_results(context, message):
    if not message:
        raise AssertionError('expected message missing')
